virttest/test_setup.py

```diff
@@ -1362,13 +1375,15 @@ def remove_driver(self, driver=None):
         if ARCH == 'ppc64le' and driver == 'mlx5_core':
             pf_devices = self.get_pf_mlx()
             logging.info("Mellanox PF devices '%s'", pf_devices)
-            for PF in pf_devices:
-                if not self.set_vf(PF):
-                    return False
             cmd = "rmmod mlx5_ib;modprobe -r mlx5_core;modprobe mlx5_ib"
-        if process.system(cmd, ignore_status=True, shell=True):
-            logging.debug("Failed to remove driver: %s", driver)
-            return False
+            if process.system(cmd, ignore_status=True, shell=True):
+                logging.debug("Failed to remove driver: %s", driver)
+                return False
+        elif ARCH == 'x86_64':
+            pf_devices = self.get_pf_ids()
+        for PF in pf_devices:
+            if not self.set_vf(PF):
```

Comment:
At this point, for `ARCH == 'ppc64le'` driver would have been removed and we cannot echo 0 to sriov_numvfs, please move it ahead.